server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1146,6 +1146,9 @@ private void closeConsistencyCheck() {
         String msg = "Data files in " + extent + " differ from in-memory data "
             + tabletMeta.getFilesMap() + " " + getDatafileManager().getDatafileSizes();
         log.error(msg);
+        getDatafileManager().logTransactions();
```

Comment:
Should I clear the transactions after loggin them here?  Thoughts?

Reply:
I think this question relates to [my earlier design suggestion](https://github.com/apache/accumulo/pull/3727#issuecomment-1699764785) wherein I suggest retaining logs only since the last successful consistency check. I think you're question here is similar in that you're asking if we should clear the transactions, once they are logged, in the case of inconsistency?

Forgive me if I'm a little confused about the details of your current implementation. I haven't been able to keep up with this. I did see that you removed it as a blocker, since the underlying issue that motivated this seems to have been fixed. That implies we could do a 2.1.3 release without this, and spend more time polishing this up after. I'll see what I can do to bump this issue back to the top of my priority list, even if it gets bumped past 2.1.3, as I'm very interested in giving this a thorough review and seeing the direction this ticket took after it dropped off my radar.

Reply:
+1

Reply:
I agree with only clearing after we are deemed consistent.  Just checking that this was the general consensus.